### PR TITLE
hdmr.ancova() vectorization

### PR DESCRIPTION
**Pull request by eyyarbasi**
hdmr.ancova() is vectorized for performance and readability.

### CODE REVIEW

**Comment by tupui** [comment]
Thank you @eyyarbasi for the PR. @sahin-abdullah do you have time to review. At first glance it looks good and CI is ok (though not sure how trustworthy that part is.)

File: src/SALib/analyze/hdmr.py [hunk 1]
```diff
@@ -671,30 +671,38 @@ def f_test(Y, f0, Y_em, R, alpha, m1, m2, m3, n1, n2, n3, n):
...
-def ancova(Y, Y_em, V_Y, R, n):
-    """Analysis of Covariance."""
+def ancova(Y, Y_em, V_Y, n, R):
+    """
+    Perform Analysis of Covariance (ANCOVA) on model and emulator outputs.
+
+    Returns:
+    tuple: A tuple containing sensitivity indices (S),
+            of structural contributions (S_a),
+            and of correlative contributions (S_b).
```
Comment (ConnectedSystems) [inline src/SALib/analyze/hdmr.py:681]: Standardize docstring format

```suggestion
    Returns
    --------
    tuple : A tuple containing sensitivity indices (S),
              of structural contributions (S_a) and of correlative contributions (S_b).
```

File: src/SALib/analyze/hdmr.py [hunk 2]
```diff
@@ -671,30 +671,38 @@ def f_test(Y, f0, Y_em, R, alpha, m1, m2, m3, n1, n2, n3, n):
...
+    # Structural contribution of jth term ( = Eq. 20 of Li et al )
+    S_a = np.var(Y_em, axis=0) / V_Y
 
-        # Correlative contribution of jth term  ( = Eq. 21 of Li et al )
-        S_b[j] = C[0, 1] / V_Y
+    # Correlative contribution of jth term ( = Eq. 21 of Li et al )
+    cov_matrix = np.empty((m, m))
+    for i in range(m):
+        cov_matrix[i] = np.cov(Y_em[:, i], Y0_minus_Y_em[:, i])[0, 1]
+    S_b = cov_matrix.diagonal() / V_Y
```
Comment (ConnectedSystems) [inline src/SALib/analyze/hdmr.py:705]: I'd prefer a new line before L705 for clarity.

**Review by ConnectedSystems — APPROVED** [review]
Requesting some minor changes but this looks good to me.

Thank you @eyyarbasi !

**Comment by willu47** [comment]
Can this be merged? @ConnectedSystems @tupui

File: src/SALib/analyze/hdmr.py (hunk 2 above)
Comment (tupui) [inline src/SALib/analyze/hdmr.py:705]:
```suggestion
        cov_matrix[i] = np.cov(Y_em[:, i], Y0_minus_Y_em[:, i])[0, 1]

    S_b = cov_matrix.diagonal() / V_Y
```

**Review by tupui — APPROVED** [review]
Looks like it yes 😃 Thanks again @eyyarbasi and sorry for the delay getting this in. It went through the cracks.

**Pull request merged**